src/protocols/secure_channel/SessionEstablishmentDelegate.h

```diff
@@ -25,13 +25,27 @@
 
 #pragma once
 
+#include <protocols/secure_channel/PairingSession.h>
 #include <system/SystemPacketBuffer.h>
 #include <transport/Session.h>
 #include <transport/raw/MessageHeader.h>
 #include <transport/raw/PeerAddress.h>
 
 namespace chip {
 
+enum class SessionState : uint8_t
+{
+    kUndefined              = 0,
```

Comment:
I think we should have an explicit "not in key exchange" state for now, which we can use in the places where we are not coming from OnSessionEstablishmentError.

OnSessionEstablishmentError could also use that state for the "error in kInitialized" state...